File: Client/src/app/shared/models/tag.model.ts
export interface ITag{
    id: number;
    index: number;
    imageId: string;
    classId: number;
    quantityCheckId: number;
    top: number;
    left: number;
    width: number;
    height: number;
}

export interface ITagForAdd{
    imageId:number;
    classId:number;
    top: number;
    left: number;
    width: number;
    height: number;
}

export interface ITagForUpdate{
    id: number;
    imageId:number;
    classId:number;
    top: number;
    left: number;
    width: number;
    height: number;
}

export class Tag implements ITag{
    constructor(
        public id: number = -1, 
        public index: number,
        public imageId: string, 
        public classId: number, 
        public quantityCheckId: number = -1, 
        public top: number, 
        public left: number,
        public width: number,
        public height: number){

    }
}

export class TaggedTimeUser{
    constructor(
        public userId:number,
        public taggedTime: number
    ){ }
}

export class DataUpdate{
    constructor(
        public userId: number, 
        public tags:Tag[], 
        public taggedTime: number,
        public excluseAreas:ExcluseArea[],
        public ignored: boolean =false
    ){}
}

export class ExcluseArea {
    constructor(
        public name: string = null,
        public paths: Coordinate[] = []
    ){}
}

export class Coordinate{
    constructor(public x:number,public y:number){}
}

File: ApiServer/Model/views/TagModel.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ApiServer.Model.views
{
    public static class TagModel
    {
        public class TagForView
        {
            public int Id { get; set; }
            public int Index { get; set; }
            public Guid ImageId { get; set; }
            public int ClassId { get; set; }
            public double Top { get; set; }
            public double Left { get; set; }
            public double Width { get; set; }
            public double Height { get; set; }
            public int QuantityCheckId { get; set; }
        }

        public class TagForAddOrUpdate
        {
            public int Id { get; set; }
            public int Index { get; set; }
            public Guid ImageId { get; set; }
            public int ClassId { get; set; }
            public double Top { get; set; }
            public double Left { get; set; }
            public double Width { get; set; }
            public double Height { get; set; }
            public double WidthPixel { get; set; }
            public double HeightPixel { get; set; }
        }

        public class TagForUpdate
        {
            public int Id { get; set; }
            public int Index { get; set; }
            public Guid ImageId { get; set; }
            public int ClassId { get; set; }
            public double Top { get; set; }
            public double Left { get; set; }
            public double Width { get; set; }
            public double Height { get; set; }
            public double WidthPixel { get; set; }
            public double HeightPixel { get; set; }
        }

        public class DataUpdate{
            public long UserId { get; set; }
            public IEnumerable<TagForUpdate> Tags { get; set; }
            public IEnumerable<ExcluseArea> ExcluseAreas { get; set; }
            public double TaggedTime { get; set; }
            public bool Ignored { get; set; }
        }

        public class ExcluseArea
        {
            public Coordinate[] Paths { get; set; }
        }

        public class Coordinate
        {
            public float X { get; set; }
            public float Y { get; set; }
        }
    }
}
